### PR TITLE
Explain in wxPlatformInfo docs that it means bitness when it says architecture

### PR DESCRIPTION
**Pull request by lanurmi**
In the context of wxPlatformInfo, 'architecture' means either '32 bit' or
'64 bit'. Anywhere outside the context of wxPlatformInfo, this concept is
known as 'bitness'.

For the rest of the world, 'architecture' generally refers to the CPU
instruction set architecture, i.e. something like x86, x86_64, arm64, whereas
'operating system architecture' refers to the design of separating kernel
space, user space, etc.

### CODE REVIEW

File: interface/wx/platinfo.h [hunk 1]
```diff
@@ -322,7 +322,7 @@ class wxPlatformInfo
     //@{
 
     /**
-        Returns the architecture ID of this wxPlatformInfo instance.
+        Returns the bitness ID of this wxPlatformInfo instance.
```
Comment (vadz) [inline interface/wx/platinfo.h:325]: Can you really say "bitness ID"?

**Review by vadz — COMMENTED** [review]
Thanks, I agree that calling this "architecture" is not ideal, but calling it just "bitness" is confusing as it has nothing to do with the actual name in the header. Could we switch to "architecture bitness" or "number of bits in the architecture" in the docs instead? IMHO this would be more clear, what do you think?

**Comment by lanurmi** [comment]
> Thanks, I agree that calling this "architecture" is not ideal, but calling it just "bitness" is confusing as it has nothing to do with the actual name in the header. Could we switch to "architecture bitness" or "number of bits in the architecture" in the docs instead?

"Architecture bitness" sounds fine yet still compact.

I was also thinking about renaming every symbol with "arch" in its name to something with "bitness". Or rather duplicating functions/types/constants and deprecating the old ones. This would open up the possibility to some day have some actual CPU architecture related info in wxPlatformInfo without confusing everybody.

> Can you really say "bitness ID"?

It's not ideal; though IMHO taken out of context it makes as much or as little sense as "architecture ID".

**Comment by lanurmi** [comment]
PR amended. Added a commit that renames and deprecates anything having "arch" in its name. Feel free to cherry-pick acceptable modifications.

**Comment by vadz** [comment]
Thanks for the update! Looks good to me, but we need to add `@since 3.1.5` to all the "new" enums/functions: even if they're just renamed from the existing ones, we should still indicate that they can't be used in the code which targets 3.0 compatibility, for example.

**Comment by vadz** [comment]
Thanks again! I'll merge this when the CI builds finish.

**Pull request merged**